### DISCUSSION TITLE
Fullstack .NET - HttpClientFactory - Modo de instanciação do HttpClient

### DISCUSSION THREAD

**Discussion by thomasravache** [balta-io/forum#685] [Tira Dúvidas] [answered]
Olá, apenas uma dúvida simples... Sempre que vejo o uso do _HttpClientFactory_, vejo que a criação de cada _HttpClient_ fica sempre na chamada de cada **método**, e no caso da utilização do _AccountHandler_ demonstrado na aula, a criação do _HttpClient_ foi feita em uma **variável** da própria classe. Gostaria de saber o por quê de utilizar dessa forma e quais são as diferenças e impactos de fazer a criação do _HttpClient_ diretamente em uma **variável** da própria classe em detrimento de efetuar a criação de cada _HttpClient_ em cada **método** da classe.

No aguardo, obrigado.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @thomasravache, como vai?

A criação no método construtor, como mostro no curso, permite o reuso do HttpClient em diversos métodos.
Desta forma, não é necessário criar um para cada método!
